src/DeviceInput/Implementations/nativeDeviceInputWrapper.ts

```diff
@@ -0,0 +1,77 @@
+import { Observable } from "../../Misc/observable";
+import { DeviceEventFactory } from "../Helpers/eventFactory";
+import { DeviceType } from "../InputDevices/deviceEnums";
+import { IDeviceEvent, IDeviceInputSystem, INativeInput } from "../Interfaces/inputInterfaces";
+
+/** @hidden */
+export class NativeDeviceInputWrapper implements IDeviceInputSystem {

```

Comment:
I'd be interested as well. I don't really understand the purpose of this layer. I generally would expect just two layers dealing with input:

1. A low layer that provides the minimum contract needed to receive raw input. I would expect a very simple contract with plain callbacks and minimal state management. Really something even thinner than the current `DeviceInputSystem` ideally.
2. A higher layer that is exported as part of the actual Babylon.js API that wraps the lower level and actually makes it pleasant to use. This is where I would expect observables and possibly additional state management to allow for richer input change events, etc. This is also where I would expect all code that is common between different low level input providers. Basically the current `DeviceSourceManager`, but ideally we'd move even more of the common logic out of `DeviceInputSystem` and into `DeviceSourceManager`.

In this PR though, it seems like there is now a third layer introduced, and more disparity between different input providers (web vs native) rather than less.

Reply:
So this PR was created to address the immediate need to fix the event creation issue in the InputManager.  It is more of an intermediate step and the code will be refactored further to clean up and streamline the input.

Reply:
The purpose of this PR is to address the event creation issue in the InputManager.  This code will ultimately be refactored further to clean up the Input process